lib/chat_web/live/main_live/index.ex

```diff
@@ -390,56 +413,161 @@ defmodule ChatWeb.MainLive.Index do
     )
   end
 
-  defp allow_chunked_upload(socket, type, opts) do
+  defp allow_file_upload(socket) do
     socket
-    |> allow_upload(type,
-      auto_upload: true,
-      max_file_size: 102_400_000_000,
+    |> allow_upload(:file,
       accept: :any,
-      max_entries: Keyword.get(opts, :max_entries, 1),
+      auto_upload: true,
       external: &chunked_presign_url/2,
+      max_entries: 2000,
+      max_file_size: 102_400_000_000,
       progress: &handle_chunked_progress/3
     )
+    |> assign(:uploads_metadata, %{})
   end
 
   def chunked_presign_url(entry, socket) do
-    {key, secret} = ChunkedFiles.new_upload()
-    socket = start_chunked_upload(socket, entry, key, secret)
+    upload_key = get_upload_key(entry, socket)
+
+    {next_chunk, key, secret} =
+      case UploadIndex.get(upload_key) do
+        nil ->
+          {key, secret} = ChunkedFiles.new_upload()
+          add_upload_to_index(socket, upload_key, key, secret)
+          {0, key, secret}
+
+        %Upload{} = upload ->
+          UploadIndex.delete(upload_key)
+          add_upload_to_index(socket, upload_key, upload.key, upload.secret)
+          next_chunk = ChunkedFiles.next_chunk(upload.key)
+          {next_chunk, upload.key, upload.secret}
+      end
+
+    {socket, uploader_data} =
+      start_chunked_upload(socket, entry, upload_key, key, secret, next_chunk)
+
     link = Helpers.upload_chunk_url(ChatWeb.Endpoint, :put, key)
 
-    {:ok, %{uploader: "UpChunk", entrypoint: link}, socket}
+    uploader_data =
+      Map.merge(%{uploader: "UpChunk", entrypoint: link, uuid: entry.uuid}, uploader_data)
+
+    {:ok, uploader_data, socket}
+  end
+
+  defp get_upload_key(%UploadEntry{} = entry, %{assigns: %{my_id: id}} = socket) do
+    destination =
+      socket
+      |> file_upload_destination()
+      |> Jason.encode!()
+      |> Base.encode64()
+
+    "#{id}:#{destination}:#{entry.client_relative_path}/#{entry.client_name}:#{entry.client_type}:#{entry.client_size}:#{entry.client_last_modified}"
+    |> Base.encode64()
```

Comment:
This way we have unencrypted data in DB telling which user uploads what file

Adding `Utils.hash` will mangle data